tests/test_datastores.py

```diff
@@ -94,3 +101,46 @@ def test_parse_url_preserve_case():
     expected_endpoint = 'Hedi'
     _, endpoint, _ = mlrun.datastore.datastore.parse_url(url)
     assert expected_endpoint, endpoint
+
+
+def test_forbidden_file_access(monkeypatch):
+    class MockV3ioClient:
+        def __init__(self, *args, **kwargs):
+            pass
+
+        def get_container_contents(self, *args, **kwargs):
+            raise RuntimeError('Permission denied')
+
+    def mock_get(*args, **kwargs):
+        mock_forbidden_response = Mock()
+        mock_forbidden_response.status_code = status.HTTP_403_FORBIDDEN
+        mock_forbidden_response.raise_for_status = Mock(
+            side_effect=requests.HTTPError('Error', response=mock_forbidden_response)
+        )
+        return mock_forbidden_response
+
+    monkeypatch.setattr(requests, "get", mock_get)
+    monkeypatch.setattr(requests, "head", mock_get)
+    monkeypatch.setattr(v3io.dataplane, "Client", MockV3ioClient)
+
+    store = mlrun.datastore.datastore.StoreManager(
+        secrets={'V3IO_ACCESS_KEY': 'some-access-key'}
+    )
+
+    with pytest.raises(AccessForbiddenError) as forbidden_exc:
+        obj = store.object('v3io://some-system/some-dir/')
+        obj.listdir()
+
+    assert forbidden_exc.value.response.status_code == status.HTTP_403_FORBIDDEN
```

Comment:
let's differentiate between the 2 tests:
1. here - we test data stores - therefore we shouldn't care too much about status codes, we want to verify the right error is being raised
2. in the API tests - test that the returned error is being translated into the right status code in the response

Reply:
IMO cover the 404 code paths as well while we're at it

Reply:
From looking into the FastAPI `TestClient` facilities, they don't support mocking and patching/mocking code that isn't specifically injected into the endpoints as a dependency...
This means, for testing this code via the API, I'd need an actual v3io system up and running, as I can't emulate it's responses.
This is why the test is in the sdk and not the api.
Once the system tests are finalized I can add tests there which use the real v3io.